src/EFCore.PG/ValueGeneration/Internal/NpgsqlValueGeneratorSelector.cs

```diff
@@ -104,6 +104,6 @@ public override bool TrySelect(IProperty property, ITypeBase typeBase, out Value
         => property.ClrType.UnwrapNullableType() == typeof(Guid)
             ? property.ValueGenerated == ValueGenerated.Never || property.GetDefaultValueSql() is not null
                 ? new TemporaryGuidValueGenerator()
-                : new GuidValueGenerator()
+                : new NpgsqlUUid7ValueGenerator()
```

Comment:
**Suggestion:** I think we should also have a test that confirms that this is now the standard behavior when working with a `Guid` that allows value generation. That particular test could simply check the version indicator, as far as I'm concerned.

Reply:
The current NpgsqlUuid7ValueGenerator_creates_uuidv7 test as it is in the PR tests the generator, but I think it should be testing NpgsqlValueGeneratorSelector.cs, as the test class implies and as other tests do. In other words, like in the other tests, the test can set up a model with a Guid key property, then get a value generated for that property and assert that it's what we want. That would confirm that it's the standard behavior.

Reply:
Having said all that, the combination of the two tests should indeed be sufficient - above you make sure that your generator is used by default, and below you make sure that that it produces the right kind of Guids.

@Timovzl or do you still see a problem here?

Reply:
Agreed, either way would assert the desired combination of factors.

@ChrisJollyAU Could you either add a test that confirms UUIDv7 as the default for a `Guid`, or adjust the existing test to test the selector for a `Guid` rather than a direct invocation? (If in doubt, I'd lean towards the former, to separate responsibilities: `Guid` should get a UUIDv7 by default; and any custom implementation we use should behave as expected when it comes to the details.)

Reply:
See my comments above about the current two tests probably being sufficient. Not against more testing, but not sure it adds value.

Reply:
@roji, @ChrisJollyAU, sorry, I misunderstood. Saw only one new test, but hadn't realized that we were already asserting that `Guid` was handled by the new generator: `AssertGenerator<NpgsqlUuid7ValueGenerator>("Guid");`

That resolves this comment. :)

Reply:
Just ran through the tests locally and found another place that uses the `AssertGenerator` which is now fixed